lib/index.js

```diff
@@ -177,6 +173,27 @@ exports.request = function (method, url, options, callback, _trace) {
         req.write(options.payload);
     }
 
+    // Custom abort method to detect early aborts
+
+    var _abort = req.abort;
+    var aborted = false;
+    req.abort = function () {
```

Comment:
That is more of a question for node. It is part of the public api, and I guess they have a reason for it to be there. See http://nodejs.org/api/http.html#http_request_abort. It is inherently also part of the wreck api, with test cases to support it. Also, there is no way to remove this.

My primary concern is that without this (or similar) patch, users can experience obscure timing related bugs when using abort. It is quite bad form to have a callback that is not always called.

I definitely agree that it sucks to do this, and I ultimately think that this should be solved in node. Until then I think we have 3 options:

 1. Use this patch
 1. Drop the abort tests & document divergent abort behavior
 1. Drop 0.12 support

Finally, there is a small possibility that I'm wrong about the missing notification and it can be solved without a hack. You are welcome to look for it.